datalad/core/distributed/push.py

```diff
@@ -438,131 +438,48 @@ def _push(dspath, content, target, data, force, jobs, res_kwargs, pbars,
-
     # cache repo type
     is_annex_repo = isinstance(ds.repo, AnnexRepo)
-
-    # TODO prevent this when `target` is a special remote
-    # (possibly redo) a push attempt to figure out what needs pushing
-    # do this on the main target only, and apply the result to all
-    # dependencies
-    try:
-        if _target:
-            # only do it when an explicit target was given, otherwise
-            # we can reuse the result from the auto-probing above
-            wannabe_gitpush = repo.push(
-                remote=target,
-                git_options=['--dry-run'])
-    except Exception as e:
-        lgr.debug(
-            'Dry-run push to check push configuration failed, '
-            'assume no configuration: %s', e)
-        wannabe_gitpush = []
-    refspecs2push = [
-        # if an upstream branch is set, go with it
-        p['from_ref']
-        if ds.config.get(
-            # refs come in as refs/heads/<branchname>
-            # need to cut the prefix
-            'branch.{}.remote'.format(p['from_ref'][11:]),
-            None) == target and ds.config.get(
-                'branch.{}.merge'.format(p['from_ref'][11:]),
-                None)
-        # if not, define target refspec explicitly to avoid having to
-        # set an upstream branch, which would happen implicitly from
-        # a users POV, and may also be hard to decide when publication
-        # dependencies are present
-        else '{}:{}'.format(p['from_ref'], p['to_ref'])
-        for p in wannabe_gitpush
-        # TODO: what if a publication dependency doesn't have it yet
-        # should we not attempt to push, because the main target has it?
-        if 'uptodate' not in p['operations'] and (
-            # cannot think of a scenario where we would want to push a
-            # managed branch directly, instead of the corresponding branch
-            'refs/heads/adjusted' not in p['from_ref'])
-    ]
-    # TODO this is not right with managed branches
-    active_branch = repo.get_active_branch()
-    if active_branch and is_annex_repo:
-        # we could face a managed branch, in which case we need to
-        # determine the actual one and make sure it is sync'ed with the
-        # managed one, and push that one instead. following methods can
-        # be called unconditionally
-        repo.localsync(managed_only=True)
-        active_branch = repo.get_corresponding_branch(
-            active_branch) or active_branch
-
-    if not refspecs2push and not active_branch:
-        # nothing was set up for push, and we have no active branch
-        # this is a weird one, let's confess and stop here
-        # I don't think we need to support such a scenario
-        if not active_branch:
+    # handling pure special remotes is a lot simpler
+    target_is_git_remote = repo.config.get(
+        f'remote.{target}.url', None) is not None
+    # TODO would is be useful to also check whether the
+    # target is set 'annex-ignore' right here?
```

Comment:
not sure... it could be a git remote with annex-ignore set (github) or not (ssh) -- doesn't change the fact that it is a  remote git can interact with (I guess).  If I got it right:
```suggestion
```